### PR TITLE
#128 add support of passing multiple options to settings

### PR DESCRIPTION
**Pull request by TaPO4eg3D**
Specifically, add support of multiple languages for `set spelllang`

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -579,13 +579,21 @@ ipcMain.on("download-list-request", (e, action, downloadId) => {
 ipcMain.on("set-permissions", (_, permissionObject) => {
     permissions = permissionObject
 })
-ipcMain.on("set-spelllang", (_, lang) => {
+ipcMain.on("set-spelllang", (_, langOptions) => {
+    langOptions = new Set(
+        langOptions.split(",").map(lang => {
+            if (lang === "system") {
```
Comment (Jelmerro) [inline app/index.js:585]: I like this change, seems much more accurate than setting an empty list (which worked on some systems, but also resulting in no language being checked on others).

File: app/index.js [hunk 2]
```diff
@@ -579,13 +579,21 @@ ipcMain.on("download-list-request", (e, action, downloadId) => {
...
+    )
+
     sessionList.forEach(ses => {
-        if (lang === "system") {
-            session.fromPartition(ses).setSpellCheckerLanguages([])
-        } else {
-            session.fromPartition(ses).setSpellCheckerLanguages([lang])
-        }
+        session.fromPartition(ses).setSpellCheckerLanguages(
+            Array.from(langOptions)
```
Comment (Jelmerro) [inline app/index.js:595]: Why do we need Array.from here? Can't we just build an array to begin with? I don't see the function of building a new set, if we are only going to use it as an array.

File: app/js/settings.js [hunk 3]
```diff
@@ -120,6 +120,9 @@ const listLike = [
     "searchwords",
     "startuppages"
 ]
+const canAcceptMultipleOptions = new Set([
```
Comment (Jelmerro) [inline app/js/settings.js:123]: I would propose the undo the changes to this file, and reimplement them similar to other like-like settings. You can follow the example of `favoritepages` and simply check each element in the list to be a valid language or "system", instead of implementing this new system just for this setting.

File: app/js/settings.js [hunk 4]
```diff
@@ -202,17 +205,28 @@ const init = () => {
...
+    if (canAcceptMultipleOptions.has(setting)) {
+        settingValues = value.split(",")
+    }
+
+    for (const settingValue of settingValues) {
+        const valid = optionList.includes(settingValue.trim())
         if (!valid) {
             const lastOption = optionList.pop()
             const text = `'${optionList.join("', '")}' or '${lastOption}'`
             UTIL.notify(`The value of setting '${setting}' can only be one of:`
```
Comment (Jelmerro) [inline app/js/settings.js:222]: If were going to implement this similar to "favoritepages", it also gives us the option to make this warning more specific and detailed, instead of showing this generic warning.

**Review by Jelmerro — COMMENTED** [review]
It's great to see that you are implementing your own suggested feature right away, though I do have a few suggestions to make it even better.

**Comment by Jelmerro** [comment]
I have updated the base branch to the new v4 one. You might also want to add your change to the CHANGELOG in the PR, though I can also do that if you don't want to.

File: app/index.js (hunk 2 above)
Comment (TaPO4eg3D) [reply app/index.js:595]: I started with a Set because I wanted to get rid of duplicates like `set spelllocale=en_US, system`. If the system already has the default language as "en_US" it would be duplicated. I don't know how Electron would process it but I rather not risk it. The same applies to passing the Set to Electron function

File: app/js/settings.js (hunk 3 above)
Comment (TaPO4eg3D) [reply app/js/settings.js:123]: Thanks for pointing out the place where I should start to look. It indeed seemed for me like an unnecessary big change for this kind of feature

File: app/index.js (hunk 2 above)
Comment (Jelmerro) [reply app/index.js:595]: Makes sense, but I would move that to happen when setting the option, so that duplicates aren't stored at all, instead of just not being used here.

**Comment by Jelmerro** [comment]
I was experimenting with removing the requirement of opening new tabs when changing the "spell" setting, and while doing that I also implemented support for passing multiple languages to "spelllang". As such, I'm closing this PR and want to apologize for implementing this myself instead of giving you the time to do it. I do want to thank you for suggesting this feature and taking the time to implement it in a PR, it has helped a lot with deciding how to approach this correctly, so thanks!
